### DISCUSSION TITLE
Type annotation for `dict[str, str]`

### DISCUSSION THREAD

**Discussion by scolobb** [hylang/hy#2613] [Q&A] [answered]
Hello,

I want to write in Hy the equivalent of the following Python snippet:
```python
type MyType = dict[str, str]
```

I found I could achieve something very close with `get`:
```hy
(deftype MyType1 (get list str))
```
```python
type MyType1 = list[str]
```

However, when I add a third argument to `get`, it gives me `dict[str][str]` instead of `dict[str, str]`:
```hy
(deftype MyType2 (get dict str str))
```
```python
type Type2 = dict[str][str]
```

My current workaround is to use `py`:
```hy
(deftype Marking (py "dict[str, str]"))
```
but it would be so nice to avoid dropping through to Python for this apparently simple type.

Is there a way to write a Hy snippet which would translate to `dict[str, str]`, without relying on `py`?

**Comment by Kodiologist** [comment, marked as answer]
If you write `dict[str, str]` with parentheses as `dict[(str, str)]`, it's a little clearer that this is just a tuple subscript of `dict`, which in Hy is `(get dict #(str str))`. See also [Hyrule's `of`](http://hylang.org/hyrule/doc/v0.7.0#hyrule.of).

**Reply by Kodiologist** [reply]
Typecheckers may also treat `(get dict [str str])`  the same as `(get dict #(str str))`.

**Reply by scolobb** [reply]
Thank you very much @Kodiologist for your very quick answer, and for teaching me Python as well as Hy! :grin: `(get dict #(str str))` worked perfectly!

FWIW, `(get _dict_ [str str])` results in `dict[[str, str]]`, which Mypy doesn't seem to like:
```
test.py:3: error: "dict" expects 2 type arguments, but 1 given  [type-arg]
test.py:3: error: Bracketed expression "[...]" is not valid as a type  [valid-type]
```
But I obviously don't mind that at all, as I have your solution now.
